### PR TITLE
Add yaml to programming guide.

### PR DESCRIPTION
**Pull request by robertwb**
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Polber

**Comment by robertwb** [comment]
CC: @VeronicaWasson

File: website/www/site/content/en/documentation/programming-guide.md [hunk 1]
```diff
@@ -628,6 +694,47 @@ the transform itself as an argument, and the operation returns the output
...
+    - name: MyTransform
+      type: MyTransformType
+      input: ProducingTransform.output_name
+      ...
+{{< /highlight >}}
+
+{{< paragraph class="language-yaml">}}
+The `.output_name` designation can be omitted for those transforms
+with a single (non-error) output.
+{{< /highlight >}}
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:714]: This seems a bit confusing - it reads like it is typical to specify the output, when in practice this is rarely done (except error handling). What if the logic was reversed by having the example be omitting the named output, and then mentioning that transforms with multiple outputs can use the `.output_name` notation (and link to error_handling doc)

File: website/www/site/content/en/documentation/programming-guide.md [hunk 2]
```diff
@@ -46,6 +46,11 @@ The [Go SDK](https://pkg.go.dev/github.com/apache/beam/sdks/v2/go/pkg/beam) supp
 The Typescript SDK supports Node v16+ and is still experimental.
 {{< /paragraph >}}
 
+{{< paragraph class="language-yaml">}}
+Yaml is supported as of Beam 2.52, but is under active development and the most
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:50]:
```suggestion
YAML is supported as of Beam 2.52, but is under active development and the most
```

File: website/www/site/content/en/documentation/programming-guide.md [hunk 3]
```diff
@@ -93,6 +98,11 @@ include:
 * I/O transforms: Beam comes with a number of "IOs" - library `PTransform`s that
   read or write data to various external storage systems.
 
+<span class="language-yaml">
+Note that in Beam YAML `PCollection`s are either implicit (e.g. when using
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:102]: nit:
```suggestion
Note that in Beam YAML, `PCollection`s are either implicit (e.g. when using
```

File: website/www/site/content/en/documentation/programming-guide.md [hunk 4]
```diff
@@ -202,6 +220,12 @@ One can either construct one manually, but it is also common to pass an object
 created from command line options such as `yargs.argv`.
 {{< /paragraph >}}
 
+{{< paragraph class="language-yaml">}}
+Pipeline options are simply an optional yaml mapping property that is a sibling to
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:224]: nit: ```suggestion
Pipeline options are simply an optional YAML mapping property that is a sibling to
```

File: website/www/site/content/en/documentation/programming-guide.md [hunk 5]
```diff
@@ -350,6 +383,11 @@ After you've created your `Pipeline`, you'll need to begin by creating at least
 one `PCollection` in some form. The `PCollection` you create serves as the input
 for the first operation in your pipeline.
 
+<span class="language-yaml">
+In Beam YAML `PCollection`s are either implicit (e.g. when using `chain`)
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:387]: nit: ```suggestion
In Beam YAML, `PCollection`s are either implicit (e.g. when using `chain`)
```

File: website/www/site/content/en/documentation/programming-guide.md [hunk 6]
```diff
@@ -367,19 +405,29 @@ adapters](#pipeline-io). The adapters vary in their exact usage, but all of them
...
 <span class="language-py">`io.TextFileSource`</span>
 <span class="language-go">`textio.Read`</span>
 <span class="language-typescript">`textio.ReadFromText`</span>,
+<span class="language-yaml">`ReadFromText`</span>,
 for example, reads from an
-external text file and returns a `PCollection` whose elements are of type
-`String`, each `String` represents one line from the text file. Here's how you
+external text file and returns a `PCollection` whose elements
+<span class="language-java language-py language-go language-typescript">
+are of type `String`, each `String`
```
Comment (Polber) [inline website/www/site/content/en/documentation/programming-guide.md:423]:
```suggestion
are of type `String` where each `String`
```

**Review by Polber — CHANGES_REQUESTED** [review]
Left a few grammatical suggestions, but looks good overall

File: website/www/site/content/en/documentation/programming-guide.md (hunk 1 above)
Comment (robertwb) [reply website/www/site/content/en/documentation/programming-guide.md]: Good point. Done.

File: website/www/site/content/en/documentation/programming-guide.md [hunk 7]
```diff
@@ -1535,7 +1670,13 @@ automatically apply some optimizations:
 
 ##### 4.2.4.1. Simple combinations using simple functions {#simple-combines}
 
+<span class="language-yaml">
+Beam YAML has the following buit-in CombineFns: count, sum, min, max,
+mean, any, all, group, and concat.
+CombineFns from other languages can also be referenced.
```
[the remark this answers is not in the document]
Comment (robertwb) [reply website/www/site/content/en/documentation/programming-guide.md]: Hopefully we can simply graduate this soon. Added a pointer.

**Review by Polber — APPROVED** [review]

**Pull request merged**